### PR TITLE
Add Sonic as search engines

### PR DESCRIPTION
**Pull request by rickytato**
Fast, lightweight & schema-less search backend. An alternative to Elasticsearch that runs on a few MBs of RAM.

Thank you for taking the time to work on a PR for Awesome-Selfhosted!

To ensure your PR is dealt with swiftly please check the following:

- [ ] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [ ] Format your submission as follows, where `Demo` and `Clients` are optional.
  Do not add a duplicate `Source code` link if it is the same as the main link.
  Keep the short description under 250 characters and use [sentence case](https://en.wikipedia.org/wiki/Letter_case#Sentence_case)
  for it, even if the project's webpage or readme uses another capitalisation
  such as title case, all caps, small caps or all lowercase.
  `Demo` links should only be used for interactive demos, i.e. not video demonstrations.
  ``- [Name](http://homepage/) - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `License` `Language` ``
- [ ] Additions that depend on proprietary services outside the user's control must be marked `⚠`.
  ``- [Name](http://homepage/) `⚠` - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `License` `Language` ``
- [ ] Additions that are not [Free software](https://en.wikipedia.org/wiki/Free_software) must be added to `non-free.md` and have the license set to `⊘ Proprietary`
- [ ] Additions are inserted preserving alphabetical order.
- [ ] Additions are not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [awesome-analytics](https://github.com/onurakpolat/awesome-analytics), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/).
- [ ] The `Language` tag is the main **server-side** requirement for the software. Don't include frameworks or specific dialects.
- [ ] Any license you add is in our [list of licenses](https://github.com/awesome-selfhosted/awesome-selfhosted/blob/master/README.md#list-of-licenses).
- [ ] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted/pulls), including closed ones.
- [ ] Any category you are creating has the minimum requirement of 3 items.
  If not, your addition may be inserted into `Misc/Other`.
- [ ] Any software project you are adding to the list is actively maintained.
- [ ] Any software project you are adding has working installation instructions.
- [ ] The pull request title is informative, unlike "Update README.md".
  Suggested titles: "Add aaa to bbb" for adding software aaa to section bbb,
  "Remove aaa from bbb" for removing, "Fix license for aaa", etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1704,6 +1704,7 @@ A [search engine](https://en.wikipedia.org/wiki/Search_engine_(computing)) is an
 - [OpenSearch](https://opensearch.org) - Open source distributed and RESTful search engine. ([Source Code](https://github.com/opensearch-project/OpenSearch)) `Apache-2.0` `Java`
 - [Searx](https://searx.github.io/searx/) - Privacy-respecting, hackable metasearch engine. ([Demo](https://searx.space/), [Source Code](https://github.com/searx/searx)) `AGPL-3.0` `Python`
 - [sist2](https://github.com/simon987/sist2) - Lightning-fast file system indexer and search tool. ([Demo](https://sist2.simon987.net/)) `GPL-3.0` `C`
+- [sonic](https://github.com/valeriansaliou/sonic) - Sonic is a fast, lightweight and schema-less search backend. It ingests search texts and identifier tuples that can then be queried against in a microsecond's time. `MPL-2.0 license` `rust`
```
Comment (KieranRobson) [inline README.md:1707]: Please change `rust` to `Rust`

Comment (KieranRobson) [reply README.md:1707]: Please change `MPL-2.0 license` to `MPL-2.0`

**Review by KieranRobson — CHANGES_REQUESTED** [review]
Couple changes to be made

**Review by kokomo123 — APPROVED** [review]
Looks good to merge after changes were made.

**Review by nodiscc — COMMENTED** [review]
As far as I can see, there is no way to query this search engine over the network, without implementing one of the client libraries in your program?

We only list end-user applications, not development toolkits or libraries, so adding this might not be relevant.

**Review by nodiscc — CHANGES_REQUESTED** [review]
@rickytato please check the relevant boxes in the PR description. Also please address the question above.

**Review by kokomo123 — COMMENTED** [review]
Nevermind then on my review. I think this isn't fit for the awesome-selfhosted list if it's just a library.

**Pull request closed without merging**